server/src/loadtest/k6_script_vote_next.js

```diff
@@ -0,0 +1,104 @@
+import http from 'k6/http'
+import { sleep } from 'k6'
+
+const vusPerScenario = 300
+
+export let options = {
+  scenarios: {
+    vote_song: { // Start 0s (concurrently)
+      startTime: '0s',
+      exec: 'vote_song',
+      executor: 'ramping-arrival-rate', // Rate not affected by the SUT's performance, unlike with ramping-vus.
+      preAllocatedVUs: vusPerScenario, // == maxVUs when maxVUs not specified.
+      startRate: 0,
+      stages: [
+        { duration: '60s', target: 1000 },
+        { duration: '30s', target: 3000 },
+        { duration: '10s', target: 3000 },
+      ],
+    },
+    next_song: { // Start 0s (concurrently)
+      startTime: '0s',
+      exec: 'next_song',
+      executor: 'ramping-arrival-rate', // Rate not affected by the SUT's performance, unlike with ramping-vus.
+      preAllocatedVUs: vusPerScenario, // == maxVUs when maxVUs not specified.
+      startRate: 0,
+      stages: [
+        { duration: '60s', target: 1000 },
+        { duration: '30s', target: 3000 },
+        { duration: '10s', target: 3000 },
+      ],
+    },
+  },
+}
+
+// Use the sample party ("Sample Party 1" "Sample Password 1") that is preloaded by our db migrations.
+const partyID = 1 // Assume the partyID since the db migration ensures that the ID will be 1.
+// If this assumption is not correct, the server console output will show errors.
+
+// Use some random song
+const songID = 13
+
+function fetchParty() {
+  // Fetch Party: Form the post request
+  var urlFetch = 'http://localhost:3000/graphql?opName=VoteSong'
+  var payloadFetch = JSON.stringify({
+    operationName: 'VoteSong',
+    variables: { partyId: partyID, songId: songID },
+    query:
+      'mutation VoteSong($partyId: Int!, $songId: Int!) {  vote(partyId: $partyId, songId: $songId) {    id    __typename  }}',
+  })
+  var paramsFetch = {
+    headers: {
+      'content-type': 'application/json',
+    },
+  }
+
+  // Fetch Party: Make the post request
+  http.post(urlFetch, payloadFetch, paramsFetch)
+}
+
+export function vote_song() {
+  // Vote: Form the post request
+  var urlVote = 'http://localhost:3000/graphql?opName=VoteSong'
+  var payloadVote = JSON.stringify({
+    operationName: 'VoteSong',
+    variables: { partyId: partyID, songId: songID },
+    query:
+      'mutation VoteSong($partyId: Int!, $songId: Int!) {  vote(partyId: $partyId, songId: $songId) {    id    __typename  }}',
+  })
+  var paramsVote = {
+    headers: {
+      'content-type': 'application/json',
+    },
+  }
+
+  // Vote: Make the post request
+  http.post(urlVote, payloadVote, paramsVote)
+
+  fetchParty()
```

Comment:
Why do you call `fetchParty()` here (and at the end of `nextSong()`)?

Reply:
It's the behavior of our app in real usage

Reply:
So would we change this once we have Subscriptions set up?

Reply:
Yes we can make a new load test and compare the two approaches if we want. It's not a perfect comparison anyways though, since this test is general and is not intended to exactly simulate the polling approach.

We can choose to make load tests to more accurately test each optimization we do, it will just be more work